indexer/services/vulcan/src/handlers/order-update-handler.ts

```diff
@@ -145,27 +145,33 @@ export class OrderUpdateHandler extends Handler {
       stats.increment(`${config.SERVICE_NAME}.order_update_with_zero_delta.count`, 1);
       return;
     }
-    const updatedQuantums: number = await this.updatePriceLevel(updateResult, sizeDeltaInQuantums);
 
-    const perpetualMarket: PerpetualMarketFromDatabase | undefined = perpetualMarketRefresher
-      .getPerpetualMarketFromTicker(updateResult.order!.ticker);
-    if (perpetualMarket === undefined) {
-      logger.error({
-        at: 'OrderUpdateHandler#handle',
-        message: `Received order update for order with unknown perpetual market, ticker ${
-          updateResult.order!.ticker}`,
-      });
-      return;
-    }
+    if (!isIOC(updateResult.order!.order!.timeInForce)) {
```

Comment:
can we reverse this logic?
```
if (isIOC) {
  return
}
other logic
```

This reduces the code complexity

Reply:
I don't entirely agree, as this only would work for errors or if the order book calculations are the last piece of logic to run. If more logic is added after this block of code, we could miss that this earlier block has an early return and not realize the logic won't run for orders with immediate execution.